source-mysql/discovery.go

```diff
@@ -471,10 +475,50 @@ func (t *mysqlColumnType) translateRecordField(val interface{}) (interface{}, er
 			return string(bs), nil
 		}
 		return val, nil
+	case "tinyint":
+		if sval, ok := asInt64(val); ok && t.Unsigned {
+			return uint64(sval) & 0xFF, nil
+		}
```

Comment:
This seems to work, but I don't grok why we need to cast to int64 and then to uint64, instead of directly from int8 to uint8?

Reply:
Yeah, on reflection I think the `asInt64` helper is an unnecessary bit of complexity. I'll replace it with type assertions for the appropriate integer sizes in each case (int8 for `tinyint`, int16 for `smallint`, etc) and only apply a mask to `mediumint` values.